extension/src/basic-code-intel.ts

```diff
@@ -0,0 +1,35 @@
+import { Handler, Providers } from '@sourcegraph/basic-code-intel'
+import * as path from 'path'
+import * as sourcegraph from 'sourcegraph'
+
+export function initBasicCodeIntel(): Providers {
```

Comment:
(not important, maybe keep in mind for future code) Since this is only a factory function now, I'd recommend the `create` prefix instead of `init`. That's the conventional prefix used e.g. in the Node stdlib. With `init`, I personally wouldn't be sure if it might do side effects like register providers in the Sourcegraph API, while `create` communicates it's only creating an object but not causes side effects (assuming that is the case here)